setup.py

```diff
@@ -1,60 +1,261 @@
+import contextlib
 import os
 import re
-import sys
-import platform
+import shlex
 import subprocess
+import sys
 
+from distutils import log
 from setuptools import setup, Extension
-from setuptools.command.build_ext import build_ext
-from distutils.version import LooseVersion
+from setuptools.command.build_ext import build_ext as build_ext_orig
+
+# NOTE(twesterhout): distutils and setuptools use a pretty strange way of
+# handling command line arguments, where some arguments are propagated to
```

Comment:
Yes I realized that...! It is much easier to just pass environment variables, but I agree that it's ugly. It's good if you found a solution here